### PR TITLE
Refactor GCE test code

### PR DESCRIPTION
**Pull request by sawsa307**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Refactor GCE test codes.

**Which issue(s) this PR fixes**:
None

**Does this PR introduce a user-facing change?**:
```release-note
None
```

### CODE REVIEW

**Review by LiorLieberman — COMMENTED** [review]
ping me to lgtm after you rebase

File: pkg/i2gw/providers/gce/ir_converter_test.go [hunk 1]
```diff
@@ -1084,3 +959,252 @@ func Test_convertToIR(t *testing.T) {
...
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      testServiceName,
+						Namespace: testNamespace,
+						Annotations: map[string]string{
+							backendConfigKey: `{"default":"backendconfig-1"}`,
+						},
+					},
+				},
+			},
+			expectedServices: expectedServices,
```
Comment (robscott) [inline pkg/i2gw/providers/gce/ir_converter_test.go:1035]: Why do all test cases have the same expected Services? What is this test actually testing?

File: pkg/i2gw/providers/gce/ir_converter_test.go [hunk 2]
```diff
@@ -1084,3 +959,252 @@ func Test_convertToIR(t *testing.T) {
...
+					Name:      testServiceName,
+					Namespace: testNamespace,
+					Annotations: map[string]string{
+						backendConfigKey: `{"default":"backendconfig-1"}`,
+					},
+				},
+			},
+			beConfigKey:    backendConfigKey,
+			expectedName:   testBeConfigName,
+			expectedExists: true,
```
Comment (robscott) [inline pkg/i2gw/providers/gce/ir_converter_test.go:1164]: Seems like we'd need at least one test case where this is false

File: pkg/i2gw/providers/gce/ir_converter_test.go [hunk 3]
```diff
@@ -36,25 +38,33 @@ import (
...
+	testBackendConfigName = "test-backendconfig"
 
-	gPathPrefix := gatewayv1.PathMatchPathPrefix
-	gExact := gatewayv1.PathMatchExact
+	saTypeClientIP = "CLIENT_IP"
+	saTypeCookie   = "GENERATED_COOKIE"
+)
 
+var (
+	// These variables are referenced by pointers.
```
Comment (robscott) [inline pkg/i2gw/providers/gce/ir_converter_test.go:52]: Can simplify a lot of this with a func like https://github.com/kubernetes-sigs/gateway-api/blob/b5e115d06513555c7bc81bf7535f896105aac450/pkg/test/cel/main_test.go#L60-L62.

**Review by robscott — COMMENTED** [review]
Thanks @sawsa307! Just a few questions, otherwise LGTM

File: pkg/i2gw/providers/gce/ir_converter_test.go (hunk 1 above)
Comment (sawsa307) [reply pkg/i2gw/providers/gce/ir_converter_test.go:1035]: It is testing: the different ways of using BackendConfig annotation on a Service, so the service doesn't need to be changed.

Comment (robscott) [reply pkg/i2gw/providers/gce/ir_converter_test.go:1035]: Generally with test cases like this I'd expect an input and expected output var. If this expected output var is not useful, what would be?

File: pkg/i2gw/providers/gce/ir_converter_test.go (hunk 2 above)
Comment (sawsa307) [reply pkg/i2gw/providers/gce/ir_converter_test.go:1139]: Added.

File: pkg/i2gw/providers/gce/ir_converter_test.go (hunk 1 above)
Comment (sawsa307) [reply pkg/i2gw/providers/gce/ir_converter_test.go]: Yes that makes sense, just updated the test case.

File: pkg/i2gw/providers/gce/ir_converter_test.go (hunk 3 above)
Comment (sawsa307) [reply pkg/i2gw/providers/gce/ir_converter_test.go]: Updated, thanks!

**Review by robscott — APPROVED** [review]
Thanks @sawsa307!

/approve

**Pull request merged**